libr/util/vector.c

```diff
@@ -56,15 +54,22 @@ static void vector_free_elems(RVector *vec) {
 	}
 }
 
+R_API void r_vector_fini(RVector *vec) {
+	r_vector_clear (vec);
+	vec->free = NULL;
+	vec->free_user = NULL;
+}
+
 R_API void r_vector_clear(RVector *vec) {
-	vector_free_elems (vec);
-	R_FREE (vec->a);
-	vec->capacity = 0;
+	if (vec) {
+		vector_free_elems (vec);
+		R_FREE (vec->a);
+		vec->capacity = 0;
+	}
 }
 
 R_API void r_vector_free(RVector *vec) {
-	vector_free_elems (vec);
-	free (vec->a);
+	r_vector_fini (vec);
```

Comment:
Will null-deref. Imo only check for null once in `_free()` and not in `_clear()` or `_fini()`.

Reply:
fixed. pls review again